### DISCUSSION TITLE
How to solve "error VBNC30737"

### DISCUSSION THREAD

**Discussion by angie-dotNet** [mono/mono#20848] [Q&A]
Hi,
I wrote my first Mono code and after compilation I got this message:
"Visual Basic.Net Compiler version 0.0.0.5943 (Mono 4.0.1 - tarball)
Copyright (C) 2004-2010 Rolf Bjarne Kvinge. All rights reserved.

vbnc : error VBNC30737: No accessible 'Main' method with an appropriate signature was found in 'Main'.
There were 1 errors and 0 warnings.
Compilation took 00:00:00.5799180"

Code is:

Option Strict Off
Option Explicit On
Imports system.console
Imports system.io

public class compiler
  shared function Main (A As Integer()) as integer
        Dim i As Integer
        Dim j As Integer

        Main = 0
        i = 0
        ReDim A(100000)

        Do While asc(Console.Read()) <> 27
            A(i) = Console.Read()
            i = i + 1
        Loop 

        For i = 0 To A.Length - 1
            For j = 0 To A.Length - 1
                If A(j) > 0 Then
                    If i <> A(j) And i < A(j) And i < Main Then Main = i
                End If
            Next
        Next

      Console.WriteLine ("  ")
      Console.Write ("The smallest positive integer (greater than 0) that does not occur in array is: ")
      Console.WriteLine (Main)
     'Return Main

  End function  
end class

Can somebody tell me where is the problem?

Thanks in advance
Ana

**Comment by lextm** [comment]
Your `Main` method is invalid. One valid example can be `Sub Main(ByVal args As String())`

**Reply by lespinozaguerrero6-png** [reply]
=== compile output ===

Visual Basic.Net Compiler version 0.0.0.5943 (Mono 4.7 - tarball)
Copyright (C) 2004-2010 Rolf Bjarne Kvinge. All rights reserved.

/box/Main.vb (1,5) : error VBNC30203: Identifier expected.
/box/Main.vb (2,9) : error VBNC30203: Identifier expected.
/box/Main.vb (3,5) : error VBNC30203: Identifier expected.
/box/Main.vb (4,2) : error VBNC30203: Identifier expected.
/box/Main.vb (5,6) : error VBNC30203: Identifier expected.
/box/Main.vb (6,13) : error VBNC30203: Identifier expected.
/box/Main.vb (7,8) : error VBNC30203: Identifier expected.
/box/Main.vb (8,5) : error VBNC30203: Identifier expected.
/box/Main.vb (9,7) : error VBNC30203: Identifier expected.
/box/Main.vb (10,6) : error VBNC30203: Identifier expected.
/box/Main.vb (11,3) : error VBNC30203: Identifier expected.
/box/Main.vb (11,50) : error VBNC30037: Symbol is not valid.
/box/Main.vb (12,5) : error VBNC30203: Identifier expected.
/box/Main.vb (13,5) : error VBNC30203: Identifier expected.
/box/Main.vb (14,7) : error VBNC30203: Identifier expected.
/box/Main.vb (15,10) : error VBNC30203: Identifier expected.
/box/Main.vb (16,10) : error VBNC30203: Identifier expected.
/box/Main.vb (17,3) : error VBNC30203: Identifier expected.
/box/Main.vb (18,6) : error VBNC30203: Identifier expected.
/box/Main.vb (19,7) : error VBNC30203: Identifier expected.
/box/Main.vb (20,5) : error VBNC30203: Identifier expected.
/box/Main.vb (21,12) : error VBNC30203: Identifier expected.
/box/Main.vb (22,12) : error VBNC30203: Identifier expected.
/box/Main.vb (23,6) : error VBNC30203: Identifier expected.
/box/Main.vb (24,2) : error VBNC30203: Identifier expected.
/box/Main.vb (25,10) : error VBNC30203: Identifier expected.
/box/Main.vb (26,4) : error VBNC30203: Identifier expected.
/box/Main.vb (27,4) : error VBNC30203: Identifier expected.
/box/Main.vb (28,8) : error VBNC30203: Identifier expected.
/box/Main.vb (29,7) : error VBNC30203: Identifier expected.
/box/Main.vb (30,3) : error VBNC30203: Identifier expected.
/box/Main.vb (31,7) : error VBNC30203: Identifier expected.
/box/Main.vb (32,4) : error VBNC30203: Identifier expected.
/box/Main.vb (33,9) : error VBNC30203: Identifier expected.
/box/Main.vb (35,2) : error VBNC30203: Identifier expected.
/box/Main.vb (36,6) : error VBNC30203: Identifier expected.
/box/Main.vb (37,11) : error VBNC30203: Identifier expected.
/box/Main.vb (38,5) : error VBNC30203: Identifier expected.
/box/Main.vb (39,13) : error VBNC30203: Identifier expected.
/box/Main.vb (41,3) : error VBNC30203: Identifier expected.
/box/Main.vb (41,27) : error VBNC30037: Symbol is not valid.
/box/Main.vb (42,6) : error VBNC30203: Identifier expected.
/box/Main.vb (43,11) : error VBNC30203: Identifier expected.
/box/Main.vb (44,2) : error VBNC30203: Identifier expected.
/box/Main.vb (45,7) : error VBNC30203: Identifier expected.
/box/Main.vb (46,15) : error VBNC30203: Identifier expected.
/box/Main.vb (47,6) : error VBNC30203: Identifier expected.
/box/Main.vb (48,4) : error VBNC30203: Identifier expected.
/box/Main.vb (49,4) : error VBNC30203: Identifier expected.
/box/Main.vb (50,10) : error VBNC30203: Identifier expected.
/box/Main.vb (51,14) : error VBNC30203: Identifier expected.
/box/Main.vb (52,11) : error VBNC30203: Identifier expected.
/box/Main.vb (54,14) : error VBNC30203: Identifier expected.
/box/Main.vb (56,5) : error VBNC30203: Identifier expected.
/box/Main.vb (57,8) : error VBNC30203: Identifier expected.
/box/Main.vb (58,14) : error VBNC30203: Identifier expected.
/box/Main.vb (59,3) : error VBNC30203: Identifier expected.
/box/Main.vb (60,2) : error VBNC30203: Identifier expected.
/box/Main.vb (61,3) : error VBNC30203: Identifier expected.
/box/Main.vb (62,11) : error VBNC30203: Identifier expected.
/box/Main.vb (64,10) : error VBNC30203: Identifier expected.
/box/Main.vb (66,5) : error VBNC30203: Identifier expected.
/box/Main.vb (67,2) : error VBNC30203: Identifier expected.
/box/Main.vb (68,5) : error VBNC30203: Identifier expected.
/box/Main.vb (69,3) : error VBNC30203: Identifier expected.
/box/Main.vb (71,7) : error VBNC30203: Identifier expected.
/box/Main.vb (74,3) : error VBNC30203: Identifier expected.
/box/Main.vb (75,6) : error VBNC30203: Identifier expected.
/box/Main.vb (76,6) : error VBNC30203: Identifier expected.
/box/Main.vb (77,7) : error VBNC30203: Identifier expected.
/box/Main.vb (78,6) : error VBNC30203: Identifier expected.
/box/Main.vb (79,13) : error VBNC30203: Identifier expected.
/box/Main.vb (80,7) : error VBNC30203: Identifier expected.
/box/Main.vb (81,8) : error VBNC30203: Identifier expected.
/box/Main.vb (82,5) : error VBNC30203: Identifier expected.
/box/Main.vb (83,2) : error VBNC30203: Identifier expected.
/box/Main.vb (84,3) : error VBNC30203: Identifier expected.
/box/Main.vb (85,12) : error VBNC30203: Identifier expected.
/box/Main.vb (86,8) : error VBNC30203: Identifier expected.
/box/Main.vb (87,14) : error VBNC30203: Identifier expected.
/box/Main.vb (88,2) : error VBNC30203: Identifier expected.
/box/Main.vb (89,9) : error VBNC30203: Identifier expected.
/box/Main.vb (91,7) : error VBNC30203: Identifier expected.
/box/Main.vb (92,8) : error VBNC30203: Identifier expected.
/box/Main.vb (93,11) : error VBNC30203: Identifier expected.
/box/Main.vb (94,14) : error VBNC30203: Identifier expected.
/box/Main.vb (95,3) : error VBNC30203: Identifier expected.
/box/Main.vb (96,3) : error VBNC30203: Identifier expected.
/box/Main.vb (98,2) : error VBNC30203: Identifier expected.
/box/Main.vb (100,13) : error VBNC30203: Identifier expected.
/box/Main.vb (101,3) : error VBNC30203: Identifier expected.
/box/Main.vb (102,3) : error VBNC30203: Identifier expected.
/box/Main.vb (103,4) : error VBNC30203: Identifier expected.
/box/Main.vb (104,53) : error VBNC30203: Identifier expected.
/box/Main.vb (105,2) : error VBNC30203: Identifier expected.
/box/Main.vb (106,3) : error VBNC30203: Identifier expected.
/box/Main.vb (107,14) : error VBNC30203: Identifier expected.
/box/Main.vb (108,6) : error VBNC30203: Identifier expected.
/box/Main.vb (109,7) : error VBNC30203: Identifier expected.
/box/Main.vb (110,5) : error VBNC30203: Identifier expected.
/box/Main.vb (111,13) : error VBNC30203: Identifier expected.
vbnc : Command line : error VBNC30041: Too many errors.

**Comment by madewokherd** [comment]
VBNC is no longer maintained. I suggest using vbc instead.
